packages/inventory/src/components/InventoryItemForm/index.tsx

```diff
@@ -148,7 +157,7 @@ const InventoryItemForm: React.FC<InventoryItemFormProps> = (props: InventoryIte
         accountabilityEndDate: accEndDate,
       });
     }
-  }, [selectedProduct, selectedDeliveryDate, form]);
+  }, [selectedProduct, selectedDeliveryDate, form, isProductChanged, isDeliveryDateChanged]);
```

Comment:
curious here , is there not some knowledge duplication here, i.e. knowing how useEffect works and then the 2 added dependency variables